pkg/cnbutils/sbom.go

```diff
@@ -0,0 +1,185 @@
+package cnbutils
+
+import (
+	"archive/tar"
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"regexp"
+
+	"github.com/pkg/errors"
+
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/anchore/syft/syft"
+	"github.com/anchore/syft/syft/sbom"
+	"github.com/anchore/syft/syft/source"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+)
+
+var cycloneDxXML = syft.FormatByID(syft.CycloneDxXMLFormatID)
+
+func MergeSBOMFiles(pattern, output, img, dockerConfigFile string, utils BuildUtils) error {
+	if dockerConfigFile != "" {
+		os.Setenv("DOCKER_CONFIG", filepath.Dir(dockerConfigFile))
+		defer os.Unsetenv("DOCKER_CONFIG")
+	}
+
+	log.Entry().Debugf("reading remote image %s", img)
+	remoteImage, err := utils.GetRemoteImageInfo(img)
+	if err != nil {
+		return err
+	}
+
+	imgConfig, err := remoteImage.ConfigFile()
+	if err != nil {
+		return err
+	}
+
+	layerSHA, exists := imgConfig.Config.Labels["io.buildpacks.base.sbom"]
+	if !exists {
+		return fmt.Errorf("image %q does not contain label \"io.buildpacks.base.sbom\"", img)
+	}
+
+	var bom *sbom.SBOM
```

Comment:
any reason why using the one from Syft and not the one from the CycloneDX project?
[github.com/CycloneDX/cyclonedx-go](https://github.com/CycloneDX/cyclonedx-go)

Reply:
Buildpacks can produce three different sbom formats: `cyclonedx`, `syft` (default) and `spdx`.
The `syft` library can work with all of them, translating one format to another. In the end, `cnbBuild` will return CycloneDX-xml file.